src/main/java/io/arlas/server/wfs/utils/WFSCheckParam.java

```diff
@@ -0,0 +1,115 @@
+            throw new WFSException(WFSExceptionCode.OPERATION_NOT_SUPPORTED, "BBOX and RESOURCEID can't be used together", "bbox,resourceid");
+        } else if (bbox != null && filter != null) {
+            throw new WFSException(WFSExceptionCode.OPERATION_NOT_SUPPORTED, "BBOX and FILTER can't be used together", "bbox,filter");
+        } else if (resourceid != null && filter != null) {
+            throw new WFSException(WFSExceptionCode.OPERATION_NOT_SUPPORTED, "RESOURCEID and FILTER can't be used together", "bbox,filter");
+        }
+
+        if (requestType != WFSRequestType.GetCapabilities) {
+            if (requestVersion == null) {
+                String msg = "Missing version parameter.";
+                throw new WFSException(WFSExceptionCode.MISSING_PARAMETER_VALUE, msg, "version");
+            }
+            VersionUtils.checkVersion(requestVersion);
+        } else {
+            if (service == null || !service.equals("WFS")) {
+                String msg = "Missing service parameter.";
+                throw new WFSException(WFSExceptionCode.MISSING_PARAMETER_VALUE, msg, "service");
+            }
+        }
+    }
+
+    public static void checkTypeNames(String collectionName,String typenames) throws WFSException {
+        if (typenames != null) {
+            if (!typenames.contains(collectionName)) {
+                throw new WFSException(WFSExceptionCode.INVALID_PARAMETER_VALUE, "FeatureType " + typenames + " not exist", "typenames");
+            }
+        }
+    }
+
+    public static void checkSrsName(String srsname) throws WFSException {
+        boolean isCrsUnSupported = false;
+        if (srsname != null) {
+            if (Arrays.asList(WFSConstant.SUPPORTED_CRS).indexOf(srsname) < 0) {
+                isCrsUnSupported = true;
+            }
+        } else {
+            isCrsUnSupported = false;
+        }
+        if (isCrsUnSupported) {
+            throw new WFSException(WFSExceptionCode.INVALID_PARAMETER_VALUE, "Invalid CRS :" + srsname, "srsname");
+        }
+    }
+
+    public static String formatValueReference(String valuereference,CollectionReferenceDescription collectionReferenceDescription) throws WFSException {
+            if (valuereference == null || valuereference.equals("")) {
+                throw new WFSException(WFSExceptionCode.INVALID_PARAMETER_VALUE, "Invalid valuereference value", "valuereference");
+            }else if(valuereference.equals("@gml:id")){
+                valuereference=collectionReferenceDescription.params.idPath;
+            }else if(!WFSCheckParam.isFieldInMapping(collectionReferenceDescription,valuereference)){
+                throw new WFSException(WFSExceptionCode.INVALID_PARAMETER_VALUE, "Invalid valuereference value, " + valuereference + " is not in queryable", "valuereference");
+            }
+            return valuereference;
+    }
+
+    public static boolean isBboxLatLonInCorrectRanges(double[] tlbr) {
```

Comment:
Duplicate code with src/main/java/io/arlas/server/utils/CheckParams.java, isn't it?

Reply:
It's not exactly the same function : 
        // valid bbox from WFS OGC SPEC = lower longitude , lower latitude , upper longitude  , upper latitude
        // valid bbox for ARLAS classic bbox = lat top,  long left,  lat bottom,  long right

Reply:
Got it